### PR TITLE
Use cache for http.client in tsdb package.

### PR DESCRIPTION
**Pull request by bergquist**
* Moved datasource caching from dataproxy into models.Datasource
* Not sure this belongs in `models` but not sure where else to put it. Having a repository/service for caching seems over kill.
* NewExecutor func can now also return error.

### CODE REVIEW

File: pkg/models/datasource_cache.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	}
+
+	return &http.Client{
+		Timeout:   time.Duration(30 * time.Second),
+		Transport: transport,
+	}, nil
+}
+
+func (ds *DataSource) GetHttpTransport() (*http.Transport, error) {
+	ptc.Lock()
```
Comment (torkelo) [inline pkg/models/datasource_cache.go]: Does this mean can never process two requests to same data source in parallel? That does not seem very good

Comment (bergquist) [inline pkg/models/datasource_cache.go]: The http request can run in parallell. But getting the *http.Transport from the dictionary can only be done one at the time. Which should be very fast.

I haven't performance tested this yet. But Ill import lots of dashboards using the script you created to make sure.

**Comment by torkelo** [comment]
this is looking good. a bit scary to merge into 4.0.2 but then again the connection pooling issue for alerting could be an issue.

**Comment by bergquist** [comment]
@torkelo yea there was a lot of changes required to do this.

**Comment by bergquist** [comment]
@torkelo want to merge this into master instead and include it in 4.1.0? I wouldn't mind. I dont think this is as critical for alerting as the data proxy since.

**Comment by bergquist** [comment]
Oh. I even targeted the master branch. My intention was to target the v4.0.x branch.

**Pull request merged**